corehq/apps/data_dictionary/util.py

```diff
@@ -62,16 +59,32 @@ def _get_current_case_types_and_properties(domain):
 
 
 def add_properties_to_data_dictionary(domain, case_type, properties):
-    case_type_obj, created = CaseType.objects.get_or_create(domain=domain, name=case_type)
-    if not created:
-        old_properties = set(CaseProperty.objects.filter(case_type=case_type_obj).values_list('name', flat=True))
-    else:
-        old_properties = set()
-    new_properties = []
-
-    for case_property in properties:
-        if case_property not in old_properties:
-            old_properties.add(case_property)
-            new_properties.append(CaseProperty(name=case_property, case_type=case_type_obj))
-
-    CaseProperty.objects.bulk_create(new_properties)
+    if properties:
+        _create_properties_for_case_types(domain, {case_type: properties})
+
+
+def _create_properties_for_case_types(domain, case_type_to_prop):
+    current_case_types, current_properties = _get_current_case_types_and_properties(domain)
+    new_case_properties = []
+
+    for case_type, props in case_type_to_prop.items():
+        if not case_type:
+            continue
+
+        try:
+            case_type_obj = current_case_types[case_type]
+        except KeyError:
+            case_type_obj = CaseType.objects.create(domain=domain, name=case_type)
+
+        for prop in props:
+            # don't add any properites to parent cases
+            if '/' in prop:
```

Comment:
They get skipped for now. The same thing happens right now if a case property is only referenced through it's child. 

It's possible that we can figure out the hierarchy with the ParentCasePropertyBuilder but we would also need to add the property to each case type that could be a parent of that case, which isn't a great solution